tools/collision_asm2bin.sh

```diff
@@ -3,12 +3,22 @@
 
 TEMP_ASM=$(mktemp collision.asm.XXX)
 TEMP_O=$(mktemp collision.o.XXX)
+OS=$(uname -s)
+
+HEAD=head
+tail=tail
+
+if [ $OS = "Darwin" ]
```

Comment:
Perhaps, instead of detecting the OS, detect if `gtail` and `ghead` exist? If those exist, it's very likely that you'd want to use them instead of `head`/`tail`, regardless of OS.
(This is basically a case of the general principle of "ask the question you need".)